XAES-256-GCM.md

```diff
@@ -0,0 +1,45 @@
+This document specifies the XAES-256-GCM authenticated encryption with additional data algorithm, based on the composition of a standard NIST SP 800-108r1 KDF and the standard NIST AES-256-GCM AEAD.
+
+The XAES-256-GCM inputs are a 256-bit key, a 192-bit nonce, a plaintext of up to approximately 64GiB, and additional data of up to 2 EiB.
+
+Unlike AES-256-GCM, the XAES-256-GCM nonce can be randomly generated for a virtually unlimited number of messages (2⁸⁰ messages with collision risk 2⁻³²). Only a 256-bit key version is specified, which provides a comfortable multi-user security margin. Like AES-256-GCM, XAES-256-GCM is not nonce misuse-resistant, nor is it key-committing. Compared to AES-256-GCM, XAES-256-GCM requires three extra invocations of the AES-256 function, but has otherwise the same performance profile.
+
+## Overview
+
+XAES-256-GCM derives a subkey for use with AES-256-GCM from the input key and half the input nonce using a NIST SP 800-108r1 KDF.
+
+The counter-based KDF (NIST SP 800-108r1, Section 4.1) is instantiated with CMAC-AES256 (NIST SP 800-38B) and the input key as *Kin*, the ASCII letter `X` (0x58) as *Label*, and the first 96 bits of the input nonce as *Context* (as recommended by NIST SP 800-38B, Section 4, point 4), to produce a 256-bit derived key.
```

Comment:
```suggestion
The counter-based KDF ([NIST SP 800-108r1, Section 4.1]) is instantiated with CMAC-AES256 ([NIST SP 800-38B]) and the input key as *Kin*, the ASCII letter `X` (0x58) as *Label*, and the first 96 bits of the input nonce as *Context* (as recommended by [NIST SP 800-38B], Section 4, point 4), to produce a 256-bit derived key.
```

I also cannot find the Section 4, point 4 reference in NIST SP 800-38B.

Reply:
It was supposed to be NIST SP 800-108r1, thanks.